datafusion/src/logical_plan/builder.rs

```diff
@@ -231,34 +231,26 @@ impl LogicalPlanBuilder {
         Ok(Self::from(table_scan))
     }
 
-    /// Apply a projection.
-    ///
-    /// # Errors
-    /// This function errors under any of the following conditions:
-    /// * Two or more expressions have the same name
-    /// * An invalid expression is used (e.g. a `sort` expression)
+    /// Apply a projection without alias.
     pub fn project(&self, expr: impl IntoIterator<Item = Expr>) -> Result<Self> {
-        let input_schema = self.plan.schema();
-        let mut projected_expr = vec![];
-        for e in expr {
-            match e {
-                Expr::Wildcard => {
-                    projected_expr.extend(expand_wildcard(input_schema, &self.plan)?)
-                }
-                _ => projected_expr
-                    .push(columnize_expr(normalize_col(e, &self.plan)?, input_schema)),
-            }
-        }
-
-        validate_unique_names("Projections", projected_expr.iter(), input_schema)?;
-
-        let schema = DFSchema::new(exprlist_to_fields(&projected_expr, input_schema)?)?;
+        Ok(Self::from(project_with_alias(
```

Comment:
I wonder if you could write this like `self.project_with_alias(expr, None)` which might be slightly cleaner